### PR TITLE
add lc 5

### PR DESCRIPTION
**Pull request by snowan**

### CODE REVIEW

**Comment by snowan** [comment]
@sunboman review

File: 2021/python/5_longest-palindromic-substr.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+when left = 0 ("a"), right = 4 ("a"), s[left] == s[right], and check "bcd" palindrome, so dp[left + 1][right - 1]
+then mark true, and track longest and start position.
+
+result substring:  s[start, start + longest]
+"""
+def longestPalindrome(s: str) -> str:
+    if not s or len(s) < 2:
+        return s
+    size = len(s)
+    dp = [[0 for j in range(size)] for i in range(size)]
```
Comment (sunboman) [inline 2021/python/5_longest-palindromic-substr.py:51]: `dp = [[0] * size] * size`

File: 2021/python/5_longest-palindromic-substr.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        dp[i][i] = 1
+    start, longest = 0, 1
+    # reverse traverse
+    for left in range(size, -1, -1):
+        for right in range(left + 1, size):
+            if s[left] == s[right]:
+                if right - left == 1 or dp[left + 1][right - 1]:
+                    dp[left][right] = 1
+                    if longest < right - left + 1:
+                        start = left
```
Comment (sunboman) [inline 2021/python/5_longest-palindromic-substr.py]: `start, longest = left, right - left + 1`

File: 2021/python/5_longest-palindromic-substr.py (hunk 1 above)
Comment (snowan) [reply 2021/python/5_longest-palindromic-substr.py:51]: This wont work. 
I don’t why with this. Init 2 dimetrial array as 0. 
But when Try to set diagonal value to 1, with this approach of init, it set all cells to 1?? Wtf
```
for i in range(size):
         dp[i][i] = 1
```
E.g two ways of init after set diagonal values
```

   0.000    0.000 
   0.000    0.000 
------------------
   0.000    0.000 
   0.000    0.000 
------------------
   1.000    1.000 
   1.000    1.000 
------------------
   1.000    0.000 
   0.000    1.000 

```

File: 2021/python/5_longest-palindromic-substr.py (hunk 2 above)
Comment (snowan) [reply 2021/python/5_longest-palindromic-substr.py]: Nice~~

**Review by sunboman — APPROVED** [review]

File: 2021/python/5_longest-palindromic-substr.py (hunk 1 above)
Comment (sunboman) [reply 2021/python/5_longest-palindromic-substr.py:51]: boo~~~

**Pull request merged**